main/data_nodes.cpp

```diff
@@ -124,6 +129,9 @@ void data_nodes_init()
     id64 += ((uint64_t)LIBRE_SOLAR_TYPE_ID) << 32;
     uint64_to_base32(id64, device_id, sizeof(device_id));
 
+    const esp_app_desc_t *description = esp_ota_get_app_description();
+    strncpy(firmware_version, description->version, 32);
```

Comment:
Safer implementation:
```suggestion
    strncpy(firmware_version, description->version, sizeof(firmware_version));
```